frontend/src/ui/pages/RegisterPage/RegisterPage.js

```diff
@@ -0,0 +1,139 @@
+import React, { useEffect, useState } from 'react'
+import './RegisterPage.scss'
+import { Formik, Form, Field, ErrorMessage } from 'formik'
+import * as Yup from 'yup'
+import { Button, Text } from '../../components'
+import { register } from '../../../ducks/UserApi'
+import { useNavigate } from 'react-router'
+import RegisterPopup from './RegisterPopup'
+import { Link } from 'react-router-dom'
+
+function RegisterPage() {
+
+	const navigate = useNavigate()
+
+	const [currentBody, setCurrentBody] = useState(0)
+	const [currentHair, setCurrentHair] = useState(0)
+	const [currentFit, setCurrentFit] = useState(0)
+	const [toggle, setToggle] = useState(false)
+
+	const [errors, setErrors] = useState([])
+    
+	const registerSchema = Yup.object().shape({
+		password: Yup.string().required('Wait! What\'s your password?'),
+		passwordConfirmation: Yup.string().required('Confirm your passowrd.').oneOf([Yup.ref('password'), null], 'Passwords don\'t match.'),
+		username: Yup.string().required('Wait! What\'s your username?')
+	})
+
+	const handleSubmit = async (values) => {
+		const avatar = {
+			'body': /^([^.]+)/.exec(Object.keys(body)[currentBody])[1],
+			'hair': /^([^.]+)/.exec(Object.keys(hair)[currentHair])[1],
+			'fit': /^([^.]+)/.exec(Object.keys(fit)[currentFit])[1]
+		}
+
+		const user = {...values, hairID: parseInt(avatar.hair),  outfitTopID: parseInt(avatar.fit), outfitBottomID: parseInt(avatar.body)}
+
+		const res = await register(user)
+		if (!res) {
+			setErrors(['Server not available'])
+		} else {
+			if (res.status === 201) {
+				setToggle(true)
+				setTimeout(() => {
+					navigate('/login')
+				}, 3000)
+			} else if (res.status === 400) {
+				setTimeout(() => {
+					setErrors([])
+				}, 4000)
+				setErrors([res.data.message, ...errors])
+			}
+		}
+	}
+
+	function importAll(r) {
+		let images = {}
+		r.keys().forEach((item) => { images[item.replace('./', '')] = r(item) })
+		return images
+	}
+    
+	// eslint-disable-next-line no-undef
+	const body = importAll(require.context('../../assets/avatar_files/body', false, /\.(png|jpe?g|svg)$/))
+	// eslint-disable-next-line no-undef
+	const fit = importAll(require.context('../../assets/avatar_files/fit', false, /\.(png|jpe?g|svg)$/))
+	// eslint-disable-next-line no-undef
+	const hair = importAll(require.context('../../assets/avatar_files/hair', false, /\.(png|jpe?g|svg)$/))
+
+	useEffect(() => {
+
+	}, [currentBody, currentFit, currentHair])
+
+
+	return (
+		<div className={`register-page ${toggle ? 'overlay' : null}`}> 
+			{(toggle) ? 
+				<RegisterPopup /> : null
```

Comment:
intstead of 
`toggle ? <RegisterPopup /> : null`
you can use:
`toggle && <RegisterPopup />`

Reply:
same for lines 74, 98 and 52 in LoginPage.js